article_store/articles/models.py

```diff
@@ -0,0 +1,84 @@
+import re
+import uuid
+
+from django.core.validators import RegexValidator
+from django.db import models
+
+
+PREVIEW = 'preview'
+PUBLISHED = 'published'
+READY = 'ready'
+
+ARTICLE_STATUSES = (
+    (PREVIEW, 'preview'),
+    (PUBLISHED, 'published'),
+    (READY, 'ready'),
+)
+
+ARTICLE_ID_FORMAT = r'^[A-Za-z0-9\-._]+$'
+
+
+class Article(models.Model):
+    id = models.CharField(primary_key=True, max_length=250,
+                          validators=[RegexValidator(regex=ARTICLE_ID_FORMAT)])
+
+    article_id_format = ARTICLE_ID_FORMAT
+    default_version = 0
+
+    def __str__(self):
+        return f'{self.id}'
+
+    @property
+    def latest_version(self) -> int:
```

Comment:
seeing how this is used, perhaps it (and `self.default_version`) could be called `version_count` instead?
Because an empty article as `0` versions when you count them, but `0` is not a valid version instead. There are no holes like a missing version 2 (according to the DELETE API behavior) so they should be equivalent.

Reply:
This came out of having to handle the `latest` keyword as a possible version value in a URL, just followed the naming from there.

I don't mind either way, I have refactored to `count` based solution.